drizzlepac/hlautils/catalog_utils.py

```diff
@@ -317,8 +331,45 @@ def __init__(self, image, param_dict, debug, tp_sources):
 
         self.sourcelist_filename = self.imgname.replace(self.imgname[-9:], self.catalog_suffix)
 
-        # Initialize attributes which get computed by class methods
-        self.bkg_used = None  # actual background used for source identification/measurement
+        # Compute catalog-independent attributes to be used for photometry computations
+        # Compute AB mag zeropoint
+        photplam = self.image.imghdu[1].header['photplam']
+        photflam = self.image.imghdu[1].header['photflam']
+        # FIX: replace the constants
+        self.ab_zeropoint = -2.5 * np.log10(photflam) - 21.10 - 5.0 * np.log10(photplam) + 18.6921
+
+        # Compute average gain
+        # FIX not four amps for all detectors
+        self.gain = self.image.imghdu[0].header['exptime'] * np.mean([self.image.imghdu[0].header['atodgna'],
+                                                                      self.image.imghdu[0].header['atodgnb'],
+                                                                      self.image.imghdu[0].header['atodgnc'],
+                                                                      self.image.imghdu[0].header['atodgnd']])
+
+        # Convert photometric aperture radii from arcsec to pixels
+        self.aper_radius_arcsec = [self.param_dict['aperture_1'], self.param_dict['aperture_2']]
+        self.aper_radius_list_pixels = []
+        for aper_radius in self.aper_radius_arcsec:
+            self.aper_radius_list_pixels.append(aper_radius/self.param_dict['scale'])
```

Comment:
self.aper_radius_list_pixels = [aper_radius/self.param_dict['scale'] for aper_radius in self.aper_radius_arcsec]

Reply:
All places were the source used the configuration scale factor have been upgraded to use the scaling factor found in the actual data.